tests/test_notify_suppliers_whether_application_made_for_framework.py

```diff
@@ -0,0 +1,202 @@
+from logging import Logger
+import mock
+from dmutils.email import DMNotifyClient
+from dmutils.email.exceptions import EmailError
+from dmapiclient import DataAPIClient
+from dmscripts.notify_suppliers_whether_application_made_for_framework import (
+    notify_suppliers_whether_application_made,
+    NOTIFY_TEMPLATES,
+)
+
+
+class TestNotifySuppliersWhetherApplicationMade:
+
+    def setup(self):
+        self.data_api_client = mock.create_autospec(DataAPIClient, instance=True)
+        self.data_api_client.get_framework.return_value = {
+            "frameworks": {
+                "slug": "g-cloud-12",
+                "name": "G-Cloud 12",
+                "frameworkExpiresAtUTC": "2017-12-31T23:59:59.999999Z",
+                "clarificationsCloseAtUTC": "2017-12-31T23:59:59.999999Z",
+                "clarificationsPublishAtUTC": "2017-12-31T23:59:59.999999Z",
+                'applicationsCloseAtUTC': "2017-12-31T23:59:59.999999Z",
+                'intentionToAwardAtUTC': "2017-12-31T23:59:59.999999Z",
+                'frameworkLiveAtUTC': "2017-12-31T23:59:59.999999Z"
+            }
+        }
+        self.data_api_client.find_framework_suppliers_iter.return_value = [
+            {"supplierId": 712345},
+            {"supplierId": 712346},
+        ]
+        self.data_api_client.export_users.return_value = {
+            # This endpoint returns some unusually formmatted keys :(
+            # See https://github.com/alphagov/digitalmarketplace-api/blob/master/app/main/views/users.py#L417
```

Comment:
This link might bit-rot as it points to `master`. Could we link perhaps to a specific hash or PR where this was introduced?

Reply:
Good spot! Will fix